reconciler/reconciler.go

```diff
@@ -749,6 +548,75 @@ func (r *Reconciler) pruneBalances(ctx context.Context, change *parser.BalanceCh
 	)
 }
 
+// skipAndPrune calls the ReconciliationSkipped
+// handler and attempts to prune.
+func (r *Reconciler) skipAndPrune(
+	ctx context.Context,
+	change *parser.BalanceChange,
+	skipCause string,
+) error {
+	if err := r.handler.ReconciliationSkipped(
+		ctx,
+		ActiveReconciliation,
+		change.Account,
+		change.Currency,
+		skipCause,
+	); err != nil {
+		return err
+	}
+
+	return r.updateQueueMapAndPrune(ctx, change)
+}
+
+// updateQueueMapAndPrune removes a *parser.BalanceChange
+// from the queueMap and attempts to prune the associated
+// *types.AccountCurrency's balances, if appropriate.
+func (r *Reconciler) updateQueueMapAndPrune(
+	ctx context.Context,
+	change *parser.BalanceChange,
+) error {
+	r.queueMapMutex.Lock()
+
+	key := types.Hash(&types.AccountCurrency{
+		Account:  change.Account,
+		Currency: change.Currency,
+	})
```

Comment:
Assuming type.Hash() is thread safe
```suggestion
	key := types.Hash(&types.AccountCurrency{
		Account:  change.Account,
		Currency: change.Currency,
	})
	
	r.queueMapMutex.Lock()
```

Reply:
Yep! It is thread-safe.